include/grpc++/impl/codegen/async_stream.h

```diff
@@ -424,6 +424,20 @@ class ClientAsyncReaderWriterInterface : public ClientAsyncStreamingInterface,
   ///
   /// \param[in] tag The tag identifying the operation.
   virtual void WritesDone(void* tag) = 0;
+
+  /// Special operation that is only available in this streaming interface
+  /// as it allows batching of multiple operations into a single function
+  /// call (and thus only a single CQ trip).
+  /// Note that \a do_read_initial_metadata is only checked when relevant since
+  /// it may be implicit otherwise. On the other hand, the other booleans
+  /// are actually required to be used correctly
+  /// \a write_request should be nullptr if nothing should be written
+  /// \a read_response should be nullptr if nothing should be read
+  /// \a status should be nullptr if not trying to get Status (not Finish-ing)
+  virtual void BatchOps(bool do_start_call, const W* write_request,
```

Comment:
What does this add (that's needed) that isn't addressed by https://github.com/grpc/proposal/blob/master/L1-c%2B%2B-stream-coalescing.md?

Reply:
That gRFC has nothing on the Read side. The FR here was to have something that has as many trips through the CQ as though it were a unary RPC. (The user in question is using a generic client to talk to a non-generic server that thinks it's getting a unary RPC.) Without bundling the Read message, the current API will still take 2 trips through the CQ rather than 1.

Reply:
But this changes non-generic API's too.

It seems like the 'right' change would be to allow the generic stub to specify arity and return different stream objects?

Reply:
Yeah, that seems like a better design. This was less painful but unfortunately has side effects that we may not want to maintain. I'll close this one and fix.